example/cpp/api.hpp

```diff
@@ -0,0 +1,93 @@
+#include <stdint.h>
+#include <stddef.h>
+#include <stdbool.h>
+#include <memory>
+
+namespace capi {
```

Comment:
followup: worth filing an issue for making all of this customizeable

Reply:
#27